saleor/checkout/tests/test_utils.py

```diff
@@ -0,0 +1,52 @@
+from decimal import Decimal
+
+import pytest
+from prices import Money, TaxedMoney
+
+from ..utils import get_taxed_undiscounted_price
+
+BASE = Money("35.00", "USD")
+
+
+@pytest.mark.parametrize(
+    "price, tax_rate, prices_entered_with_tax, result",
+    [
+        # result should not be calculated but taken from price
+        (
+            TaxedMoney(gross=Money("43.74", "USD"), net=BASE),
+            Decimal(0.25),
+            False,
+            TaxedMoney(gross=Money("43.74", "USD"), net=BASE),
+        ),
+        # result should be calculated and different from price
+        (
+            TaxedMoney(gross=Money("43.74", "USD"), net=Money("36.00", "USD")),
+            Decimal(0.25),
+            False,
+            TaxedMoney(gross=Money("43.75", "USD"), net=BASE),
```

Comment:
Why this gross is different than gross from line 23?

Reply:
It is connected to the functionality described [here](https://github.com/saleor/saleor/pull/13585/files/8368e1944d6cd2b4ad8f741962864974b5203f00#diff-2f35ab82d7f669cedfbd9a04be1c39c1a1c3ee299730d7f0470254f5cc37721fR969) that takes into consideration that if local tax calculations for undiscounted prices should be the same, the price is always taken from the tax_app.

Reply:
I suppose that since I have to explain it it is not properly described in docstring and comments so I will need to fix that (unless you have a concern regarding this whole solution)